app/src/main/java/org/wikipedia/search/SearchResultsFragment.java

```diff
@@ -578,5 +650,88 @@ private Callback callback() {
     private String getSearchLanguageCode() {
         return ((SearchFragment) getParentFragment()).getSearchLanguageCode();
     }
+
+    private void gatherSearchResultsFromAppSources() {
+        addSearchResultFromReadingLists();
+        addSearchResultFromOpenTabs();
+        addSearchResultFromHistoryEntries();
+    }
+
+    private void addSearchResultFromHistoryEntries() {
+        if (loaderManager == null) {
+            loaderManager = LoaderManager.getInstance(requireActivity()).initLoader(SEARCH_FRAGMENT_HISTORY_LOADER_ID, null, loaderCallback);
+        } else {
+            LoaderManager.getInstance(requireActivity()).restartLoader(SEARCH_FRAGMENT_HISTORY_LOADER_ID, null, loaderCallback);
+        }
+    }
+
+    private void addSearchResultFromReadingLists() {
+        List<SearchResult> searchResultsFromAppSources = new ArrayList<>();
+        List<Object> list = ReadingListBehaviorsUtil.INSTANCE.applySearchQuery(currentSearchTerm, ReadingListDbHelper.instance().getAllLists());
```

Comment:
This is a bit wasteful because it loads the entire collection of reading lists into memory, just for the purpose of finding a single page.  You could just query the database table directly, and find a single result.  The `ReadingListDbHelper` class has a function that _almost_ does what we need (`findPageInAnyList`), so you can model it from that.